src/java/fr/paris/lutece/plugins/document/service/DocumentContentService.java

```diff
@@ -774,15 +749,61 @@ public Object clone(  )
      */
     public void removeFromCache( String strDocumentId, String strPortletId )
     {
-        if ( getCache(  ) != null )
+        if ( _cache != null && _cache.isCacheEnable( ) )
         {
             String strKey = getKey( strDocumentId, strPortletId, LOCALE_FR, PortalJspBean.MODE_HTML );
 
-            getCache(  ).remove( strKey );
+            _cache.remove( strKey );
 
             strKey = getKey( strDocumentId, strPortletId, LOCALE_EN, PortalJspBean.MODE_HTML );
 
-            getCache(  ).remove( strKey );
+            _cache.remove( strKey );
         }
     }
+    
+	private static class DocumentContentCacheEntryListener<K, V> implements CacheEntryRemovedListener<K, V>, CacheEntryExpiredListener<K, V>, Serializable
```

Comment:
This implementation is not compatible with multi-instance mode: each instance has its own keyMemory. You need to analyze whether synchronization during cache insertion is truly necessary; if not, this mechanism should be removed